src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java

```diff
@@ -281,12 +281,11 @@ private static void addOldMapping() {
         if (USE_OLDMAPPING) {
             aliases.put("EST", "America/New_York");
             aliases.put("MST", "America/Denver");
-            aliases.put("HST", "Pacific/Honolulu");
         } else {
-            zones.put("EST", new ZoneInfo("EST", -18000000));
-            zones.put("MST", new ZoneInfo("MST", -25200000));
-            zones.put("HST", new ZoneInfo("HST", -36000000));
+            aliases.put("EST", "America/Panama");
+            aliases.put("MST", "America/Phoenix");
         }
+        aliases.put("HST", "Pacific/Honolulu");
```

Comment:
Looks correct to me, since "HST" to "Pacific/Honolulu" link is same for `USE_OLDMAPPING` as well as 2024b update.